src/riak_kv_wm_index.erl

```diff
@@ -456,14 +477,30 @@ handle_all_in_memory_index_query(RD, Ctx) ->
             {{error, Reason}, RD, Ctx}
     end.
 
+
+%% Return `keys', `terms', or `objects' depending on the value of
+%% `return_terms' and `return_body'
```

Comment:
Hmm. This looks familiar. I wonder if this should be in some kind of utility module since it seems to get used a bunch and trips people up.